boa/src/builtins/reflect/mod.rs

```diff
@@ -0,0 +1,374 @@
+                2,
+            )
+            .function(Self::get_prototype_of, "getPrototypeOf", 1)
+            .function(Self::has, "has", 2)
+            .function(Self::is_extensible, "isExtensible", 1)
+            .function(Self::own_keys, "ownKeys", 1)
+            .function(Self::prevent_extensions, "preventExtensions", 1)
+            .function(Self::set, "set", 3)
+            .function(Self::set_prototype_of, "setPrototypeOf", 3)
+            .property(
+                to_string_tag,
+                Reflect::NAME,
+                Attribute::READONLY | Attribute::NON_ENUMERABLE | Attribute::CONFIGURABLE,
+            )
+            .build();
+        (Self::NAME, object.into(), Self::attribute())
+    }
+}
+
+impl Reflect {
+    /// Calls a target function with arguments.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-reflect.apply
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/apply
+    pub(crate) fn apply(_: &Value, args: &[Value], context: &mut Context) -> Result<Value> {
+        let undefined = Value::undefined();
+        let target = args
+            .get(0)
+            .and_then(|v| v.as_object())
+            .ok_or_else(|| context.construct_type_error("target must be a function"))?;
+        let this_arg = args.get(1).unwrap_or(&undefined);
+        let args_list = args
+            .get(2)
+            .and_then(|v| v.as_object())
+            .ok_or_else(|| context.construct_type_error("args list must be an object"))?;
+
+        if !target.is_callable() {
+            return Err(context.construct_type_error("target must be a function"));
+        }
+        let args = args_list.create_list_from_array_like(&[], context)?;
+        target.call(this_arg, &args, context)
+    }
+
+    /// Calls a target function as a constructor with arguments.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-reflect.construct
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/construct
+    pub(crate) fn construct(_: &Value, args: &[Value], context: &mut Context) -> Result<Value> {
+        let mut target = args
+            .get(0)
+            .and_then(|v| v.as_object())
+            .ok_or_else(|| context.construct_type_error("target must be a function"))?;
+        let args_list = args
+            .get(1)
+            .and_then(|v| v.as_object())
+            .ok_or_else(|| context.construct_type_error("args list must be an object"))?;
+
+        if let Some(new_target) = args.get(2) {
+            target = new_target
+                .as_object()
+                .ok_or_else(|| context.construct_type_error("newTarget must be an object"))?;
```

Comment:
I don't think doing this assignment is spec compliant, similar to the situation with `get` and `set`. I don't think [`Construct`](https://tc39.es/ecma262/#sec-construct) is implemented (can't find the url in the repo) so this might require a bigger change.

Reply:
There is a `construct` in `GcObject`, but it does not have a `newTarget` argument. This is probably something that can be tracked in a new issue, just like `[[Get]]` and `[[Set]]`.

Reply:
Waiting for #1045 to land.

Reply:
#1045 has been merged! :D